src/app/DomainSettingsPage.qml

```diff
@@ -0,0 +1,390 @@
+            model: DomainSettingsModel
+            sortOrder: Qt.AscendingOrder
+        }
+
+        ListItem {
+            id: useragentsMenu
+            z: 3
+            // custom user agents deactivated for now
+            height: 0 //units.gu(6)
+            color: theme.palette.normal.background
+            ListItemLayout {
+                title.text: i18n.tr("Custom User Agents")
+                ProgressionSlot {}
+            }
+
+            onClicked: customUserAgentsViewLoader.active = true
+        }
+
+        ListView {
+            id: domainSettingsListView
+            anchors.topMargin: useragentsMenu.height
+            anchors.fill: parent
+            focus: true
+            model:  SortFilterModel {
+                model: domainSettingsSortedModel
+            }
+
+            ViewItems.selectMode: selectMode
+
+            delegate: ListItem {
+                id: item
+                readonly property bool isCurrentItem: item.ListView.isCurrentItem
+                readonly property string domain: model.domain
+                readonly property int userAgentId: model.userAgentId
+                height: isCurrentItem ? layout.height : units.gu(5)
+                color: isCurrentItem ? theme.palette.selected.base : theme.palette.normal.background
+
+                MouseArea {
+                    anchors.fill: parent
+                    onClicked: domainSettingsListView.currentIndex = index
+                }
+
+                SlotsLayout {
+                    id: layout
+                    width: parent.width
+
+                    mainSlot:
+
+                        Column {
+
+                        spacing: units.gu(2)
+
+                        Label {
+                            id: domainLabel
+                            width: parent.width
+                            height: units.gu(1)
+                            text: model.domain
+                            font.bold: item.ListView.isCurrentItem
+                        }
+
+                        Row {
+                            spacing: units.gu(1.5)
+                            height: units.gu(1)
+                            visible: item.ListView.isCurrentItem
+
+                            Label  {
+                                text: i18n.tr("allow custom schemes")
+                            }
+
+                            CheckBox {
+                                checked: model.allowCustomUrlSchemes
+                                onTriggered: DomainSettingsModel.allowCustomUrlSchemes(model.domain, checked)
+                            }
+                        }
+
+
+                        Row {
+                            spacing: units.gu(1.5)
+                            height: units.gu(1)
+                            visible: item.ListView.isCurrentItem
+
+                            Label  {
+                                text: i18n.tr("allow location access")
+                            }
```

Comment:
Is it appropriate to have location settings here? I think that "Settings -> Privacy and Permissions" already contains camera and microphone permissions, so it should also have these.

Reply:
I think it was done to have all domain specific settings in one place. (e.g. adding, editing, and removing them), as they are in a common data model.
would it help to put the whole category "domain settings" to "privacy and permissions" ?

Reply:
b.t.w. the microphone and camera permissions are not functional at the moment

Reply:
Moving them would make sense. A little unfortunate about mic and camera.